### PR TITLE
Improve carryover dependency handling

### LINKED ISSUE #622: [Bug]: Issues with `tc_analysis` in zppy
### What happened?

I ran into a few issues with setting up `tc_analysis` with a fully configured zppy run. 
1. Status file shows "RUNNING" after slurm error: `slurmstepd: error: *** JOB 588238 ON chr-0471 CANCELLED AT 2024-09-20T16:27:45 DUE TO TIME LIMIT ***`
2. All three e3sm_diags runs (model vs obs, model vs model, model vs model [land=only]) are waiting for `tc_analysis` even though only the model vs obs task depends on `tc_analysis`. 

### What machine were you running on?

Chrysalis

### Environment

e3sm_unified_1.10

### What command did you run?

```bash
zppy -c
```

### Copy your cfg file

```bash
[default]
input = /lcrc/group/e3sm2/ac.wlin/E3SMv3/v3.LR.historical_0051
output = /lcrc/group/e3sm2/ac.zhang40/E3SMv3/v3.LR.historical_0920
case = v3.LR.historical_0051
www = /lcrc/group/e3sm/public_html/diagnostic_output/ac.zhang40/E3SMv3_0920
partition = compute
environment_commands = "source /lcrc/soft/climate/e3sm-unified/load_latest_e3sm_unified_chrysalis.sh"
#environment_commands = "source /lcrc/soft/climate/e3sm-unified/test_e3sm_unified_1.10.0rc5_chrysalis.sh"
campaign = "water_cycle"

[climo]
active = True
#years = "0001:0100:50", "0001:0100:100"
#years = "1985:2014:30", "1985:2014:15"
years = "1985:2014:30",
walltime = "1:00:00"

  [[ atm_monthly_180x360_aave ]]
  input_subdir = "archive/atm/hist"
  mapping_file = map_ne30pg2_to_cmip6_180x360_aave.20200201.nc
  frequency = "monthly"

  [[ atm_monthly_diurnal_8xdaily_180x360_aave ]]
  input_subdir = "archive/atm/hist"
  input_files = "eam.h3"
  mapping_file = map_ne30pg2_to_cmip6_180x360_aave.20200201.nc
  vars = "PRECT"
  frequency = "diurnal_8xdaily"

  [[ land_monthly_climo ]]
  frequency = "monthly"
  input_files = "elm.h0"
  input_subdir = archive/lnd/hist
  vars = ""

[ts]
active = True
years = "1985:2014:30"
walltime = "00:50:00"


  [[ atm_monthly_180x360_aave ]]
  input_subdir = "archive/atm/hist"
  input_files = "eam.h0"
  frequency = "monthly"
  mapping_file = /home/ac.zender/data/maps/map_ne30pg2_to_cmip6_180x360_aave.20200201.nc
  vars = "FSNTOA,FLUT,FSNT,FLNT,FSNS,FLNS,SHFLX,QFLX,TAUX,TAUY,PRECC,PRECL,PRECSC,PRECSL,TS,TREFHT,CLDTOT,CLDHGH,CLDMED,CLDLOW,U,ICEFRAC,LANDFRAC,OCNFRAC,PS,CLDICE,CLDLIQ,T,AODDUST"
# Needed for mixed-phase partition
#  vars = "LANDFRAC,CLDICE,CLDLIQ,T" 
  ts_fmt = "cmip"

  [[ atm_daily_180x360_aave ]]
  input_subdir = "archive/atm/hist"
  input_files = "eam.h1"
  frequency = "daily"
  mapping_file = /home/ac.zender/data/maps/map_ne30pg2_to_cmip6_180x360_aave.20200201.nc
  # Needed for Wheeler Kiladis
  vars = "FLUT,PRECT,U850"

  [[ atm_monthly_glb ]]
  input_subdir = "archive/atm/hist"
  input_files = "eam.h0"
  frequency = "monthly"
  mapping_file = "glb"

  [[ land_monthly ]]
  input_subdir = "archive/lnd/hist"
  input_files = "elm.h0"
  frequency = "monthly"
  #mapping_file = ""
  mapping_file = map_r05_to_cmip6_180x360_aave.20231110.nc
  vars = "FSH,RH2M,LAISHA,LAISUN,QINTR,QOVER,QRUNOFF,QSOIL,QVEGE,QVEGT,SOILICE,SOILLIQ,SOILWATER_10CM,TSA,TSOI,H2OSNO,TOTLITC,CWDC,SOIL1C,SOIL2C,SOIL3C,SOIL4C,WOOD_HARVESTC,TOTVEGC,NBP,GPP,AR,HR"
  extra_vars = "landfrac"
  ts_fmt = "cmip"
#
  [[ rof_monthly ]]
  input_subdir = "archive/rof/hist"
  input_files = "mosart.h0"
  mapping_file = ""
  frequency = "monthly"
  vars = "RIVER_DISCHARGE_OVER_LAND_LIQ"
  extra_vars = 'areatotal2'

  [[ land_monthly ]]
  input_subdir = "archive/lnd/hist"
  input_files = "elm.h0"
  frequency = "monthly"
  #mapping_file = ""
  mapping_file = map_r05_to_cmip6_180x360_aave.20231110.nc
  vars = "FSH,RH2M,LAISHA,LAISUN,QINTR,QOVER,QRUNOFF,QSOIL,QVEGE,QVEGT,SOILICE,SOILLIQ,SOILWATER_10CM,TSA,TSOI,H2OSNO,TOTLITC,CWDC,SOIL1C,SOIL2C,SOIL3C,SOIL4C,WOOD_HARVESTC,TOTVEGC,NBP,GPP,AR,HR"
  extra_vars = "landfrac"
  ts_fmt = "cmip"
#
  [[ rof_monthly ]]
  input_subdir = "archive/rof/hist"
  input_files = "mosart.h0"
  mapping_file = ""
  frequency = "monthly"
  vars = "RIVER_DISCHARGE_OVER_LAND_LIQ"
  extra_vars = 'areatotal2'

  [[ lnd_monthly_glb ]]
  frequency = "monthly"
  input_files = "elm.h0"
  input_subdir = "archive/lnd/hist"
  mapping_file = "glb"
  vars = "FSH,RH2M,LAISHA,LAISUN,QINTR,QOVER,QRUNOFF,QSOIL,QVEGE,QVEGT,SOILWATER_10CM,TSA,H2OSNO,TOTLITC,CWDC,SOIL1C,SOIL2C,SOIL3C,SOIL4C,WOOD_HARVESTC,TOTVEGC,NBP,GPP,AR,HR"

[tc_analysis]
active = True
scratch = "/lcrc/globalscratch/$USER"
# Make walltime very short to reproduce this error
walltime = "00:10:00"
years = "1985:2014:30",



[e3sm_diags]
active = True
walltime = "4:00:00"
#years = "0001:0100:50", "0001:0100:100"
years = "1985:2014:30",
ts_num_years = 30
ref_start_yr = 1985
ref_final_yr = 2014
multiprocessing = True
num_workers = 8

  [[ atm_monthly_180x360_aave ]]
#  environment_commands = "source /home/ac.zhang40/y/etc/profile.d/conda.sh; conda activate edv2110"
  short_name = 'v3.LR.historical_0051'
  grid = '180x360_aave'
  reference_data_path = '/lcrc/soft/climate/e3sm_diags_data/obs_for_e3sm_diags/climatology'
  obs_ts = '/lcrc/soft/climate/e3sm_diags_data/obs_for_e3sm_diags/time-series'
  dc_obs_climo = '/lcrc/group/e3sm/public_html/e3sm_diags_test_data/unit_test_complete_run/obs/climatology'
  climo_diurnal_subsection = "atm_monthly_diurnal_8xdaily_180x360_aave"
  climo_diurnal_frequency = "diurnal_8xdaily"
  ts_daily_subsection = "atm_daily_180x360_aave"
  sets="lat_lon","zonal_mean_xy","zonal_mean_2d","polar","cosp_histogram","meridional_mean_2d","annual_cycle_zonal_mean","qbo","diurnal_cycle","zonal_mean_2d_stratosphere","aerosol_aeronet","tropical_subseasonal","tc_analysis",
#"mp_partition","aerosol_budget",
#  sets="tropical_subseasonal",
  output_format_subplot = "pdf",

  [[ lnd_monthly_mvm_lnd ]]
  # Test model-vs-model using the same files as the reference
  #environment_commands = "source /home/ac.zhang40/y/etc/profile.d/conda.sh; conda activate edv290"
  grid = 'native'
  climo_subsection = "land_monthly_climo"
  diff_title = "Difference"
  partition = "compute"
  qos = "regular"
  short_name = v3.LR.piControl
  ref_name = "20231209.v3.LR.piControl-spinup.chrysalis"
  ref_start_yr = 0051
  ref_final_yr = 0100
  ref_years = "0051-0100",
  reference_data_path = "/lcrc/group/e3sm/ac.zhang40/tests/20231209.v3.LR.piControl-spinup.chrysalis_land_diags/post/lnd/native/clim"
  run_type = "model_vs_model"
  sets = "lat_lon_land",
  short_ref_name = "20231209.v3.LR.piControl-spinup"
  swap_test_ref = False
  tag = "model_vs_model"
  ts_num_years_ref = 50

  [[atm_monthly_180x360_aave_mvm]]
  #years = "0001-0050", 
  #years = "1985-2014", 
  ref_years = "0001-0050",
  ref_start_yr = 1
  ref_final_yr = 50
  ts_num_years = 30 
  ts_num_years_ref = 10
  ts_subsection = "atm_monthly_180x360_aave"
  short_name = 'v3alpha04-COARE.piControl'
  grid = '180x360_aave'
  ref_name = '20230924.v3alpha04_trigrid.piControl.chrysalis'
  short_ref_name = 'v3alpha04-CTL.piControl'
  tag = 'v3alpha04i-COARE_vs_CTL'
  run_type = "model_vs_model"
  reference_data_path = '/lcrc/group/e3sm2/ac.xzheng/E3SMv3_dev/20230924.v3alpha04_trigrid.piControl.chrysalis/post/atm/180x360_aave/clim'
  climo_diurnal_subsection = "atm_monthly_diurnal_8xdaily_180x360_aave"
  climo_diurnal_frequency = "diurnal_8xdaily"
  climo_subsection = "atm_monthly_180x360_aave"
  sets="lat_lon","zonal_mean_xy","zonal_mean_2d","polar","cosp_histogram","meridional_mean_2d","annual_cycle_zonal_mean","qbo","diurnal_cycle","zonal_mean_2d_stratosphere","aerosol_budget"
  diff_title = 'Difference'
#  output_format_subplot = "pdf",

[ilamb]
active = True 
nodes = 8
walltime = "2:00:00"
partition = compute 
short_name = 'v3.LR.historical_0051'
#ts_land_grid = 'native'
ts_num_years = 30
years = "1985:2014:30"

[global_time_series]
active = True
experiment_name = "v3.LR.historical_0051"
figstr = "v3.LR.historical_0051"
#plots_original = "net_toa_flux_restom,global_surface_air_temperature,toa_radiation,net_atm_energy_imbalance,net_atm_water_imbalance"
plots_atm = "TREFHT,AODDUST"
plots_lnd = "FSH,RH2M,LAISHA,LAISUN,QINTR,QOVER,QRUNOFF,QSOIL,QVEGE,QVEGT,SOILWATER_10CM,TSA,H2OSNO,TOTLITC,CWDC,SOIL1C,SOIL2C,SOIL3C,SOIL4C,WOOD_HARVESTC,TOTVEGC,NBP,GPP,AR,HR"
ts_num_years = 30
walltime = "00:30:00"
years = "1985-2014",
climo_years ="1985-2014",
ts_years ="1985-2014",
moc_file = "mocTimeSeries_1985-2014.nc"
```

### What jobs are failing?

_No response_

### What stack trace are you encountering?

_No response_

### PR DESCRIPTION
**Pull request by forsyth2**
## Issue resolution
- Initially this was meant to close #622. However, #631 ended up closing that by removing the carryover dependency requirement in `e3sm_diags` entirely. That said, #622 did reveal discrepancies in how dependencies were reset (or not) amongst the different tasks. This PR creates consistent dependency handling amongst the tasks.

Select one: This pull request is...
- [x] a bug fix: increment the patch version
- [ ] a small improvement: increment the minor version
- [ ] an incompatible (non-backwards compatible) API change: increment the major version

## 1. Does this do what we want it to do?

Objectives:
- If a task should wait for a previous run of itself, that "carryover dependency" should remain respected.
- Otherwise, a task should re-establish its own dependencies, without including dependencies a previous instantiation might have required. 
- Dependencies should be listed in the `.settings` file to ease debugging (i.e., it will be obvious to tell what zppy thought the dependencies were)

Required:
- [x] Product Management: I have confirmed with the stakeholders that the objectives above are correct and complete.
  - This is the most logical design decision, based on the bug raised in #622.
- [x] Testing: I have added or modified at least one "min-case" configuration file to test this change. Every objective above is represented in at least one cfg.
- [x] Testing: I have considered likely and/or severe edge cases and have included them in testing.

## 2. Are the implementation details accurate & efficient?

Required:
- [x] Logic: I have visually inspected the entire pull request myself.
- [x] Logic: I have left GitHub comments highlighting important pieces of code logic. I have had these code blocks reviewed by at least one other team member.
 - Left comments, no block here requires review.

## 3. Is this well documented?

Required:
- [ ] Documentation: by looking at the docs, a new user could easily understand the functionality introduced by this pull request.
  - This is largely a behind-the-scenes change, enabling zppy to run faster by parallelizing more.

## 4. Is this code clean?

Required:
- [x] Readability: The code is as simple as possible and well-commented, such that a new team member could understand what's happening.
- [x] Pre-commit checks: All the pre-commits checks have passed.

### CODE REVIEW

**Comment by forsyth2** [comment]
Test output:

```
$ cd /lcrc/group/e3sm/ac.forsyth2/zppy_min_case_carryover_dependencies_output/i622-test-carryover/v3.LR.historical_0051/post/scripts

$ grep -v "OK" *status
# No output, good

$ grep -A 3 "'dependencies'" *settings
```
That gives the following dependencies:

e3sm_diags_atm_monthly_180x360_aave_model_vs_obs_1987-1988.settings: 
- climo_atm_monthly_180x360_aave_1987-1988.status
- tc_analysis_1987-1988.status

e3sm_diags_atm_monthly_180x360_aave_mvm_model_vs_model_1987-1988_vs_1985-1986.settings: 
- climo_atm_monthly_180x360_aave_1987-1988.status [BUG: carried over]
- tc_analysis_1987-1988.status [BUG: carried over]
- e3sm_diags_atm_monthly_180x360_aave_model_vs_obs_1987-1988.status [BUG; it looks like this is marked as a carryover dependency because `e3sm_diags.py` is expecting another model_vs_obs after that uses tc_analysis, e.g., 1989-1990, but the mvm run here instead gets stuck with the dependency.
- climo_atm_monthly_180x360_aave_1987-1988.status

global_time_series_1985-1995.settings: 
- ts_atm_monthly_glb_1985-1989-0005.status
- ts_atm_monthly_glb_1990-1994-0005.status
- ts_lnd_monthly_glb_1985-1989-0005.status
- ts_lnd_monthly_glb_1990-1994-0005.status
- mpas_analysis_ts_1985-1989_climo_1985-1989.status
- mpas_analysis_ts_1985-1995_climo_1990-1995.status

ilamb_1985-1988.settings:  
- ts_land_monthly_1985-1986-0002.status
- ts_land_monthly_1987-1988-0002.status
- ts_atm_monthly_180x360_aave_1985-1986-0002.status
- ts_atm_monthly_180x360_aave_1987-1988-0002.status

mpas_analysis_ts_1985-1989_climo_1985-1989.settings:
- N/A

mpas_analysis_ts_1985-1995_climo_1990-1995.settings:  
- mpas_analysis_ts_1985-1989_climo_1985-1989.status

tc_analysis_1985-1986.settings:
- N/A
 
tc_analysis_1987-1988.settings:
- tc_analysis_1985-1986.status

File: zppy/e3sm_diags.py [hunk 1]
```diff
@@ -33,10 +33,12 @@ def e3sm_diags(config, scriptDir, existing_bundles, job_ids_file):  # noqa: C901
         return existing_bundles
 
     # --- Generate and submit e3sm_diags scripts ---
-    dependencies: List[str] = []
```
[the remark this answers is not in the document]
Comment (chengzhuzhang) [reply zppy/e3sm_diags.py:36]: @forsyth2 Thank for looking into this. To start fresh dependencies list for each `e3sm_diags` sub tasks, we can just reset `dependencies = []`, when looping over tasks, i.e. under line 38 in e3sm_diags.py without touching other part of zppy.  I'm not sure why it is needed to introduce another parameter called `carried_over_dependencies`?

[the remark this answers is not in the document]
Comment (forsyth2) [reply zppy/e3sm_diags.py:36]: @chengzhuzhang we can't start a completely fresh dependencies list on each new task because there is a case where we do want to "remember" dependencies from the previous one. This happens when `tc_analysis` is a dependency. 

In https://github.com/E3SM-Project/zppy/blob/main/zppy/e3sm_diags.py:
```
                    # Due to a `socket.gaierror: [Errno -2] Name or service not known` error when running e3sm_diags with tc_analysis
                    # on multiple year_sets, if tc_analysis is in sets, then e3sm_diags should be run sequentially.
                    if "tc_analysis" in c["sets"]:
                        # Note that this line should still be executed even if jobid == -1
                        # The later tc_analysis-using e3sm_diags tasks still depend on this task (and thus will also fail).
                        # Add to the dependency list
                        dependencies.append(statusFile)
```

A socket error occurs if we try to run `e3sm_diags` in parallel on `tc_analysis`. To work around that, we simply make `e3sm_diags` run sequentially when it's working with `tc_analysis`. (`dependencies.append(statusFile)` adds the current task to the dependencies for the next task -- _this_ is the point of `carried_over_dependencies`). This is similar to how we make `mpas_analysis` run sequentially.

The unfortunate side effect is that `e3sm_diags` tasks that do _not_ deal with `tc_analysis` also end up caught in the sequential dependency chaining. And that means they don't run if an earlier task fails.

[the remark this answers is not in the document]
Comment (chengzhuzhang) [reply zppy/e3sm_diags.py:36]: I'm curious how `tc_analysis` dependency works differently than other dependencies such as climo and time-series. I'm interested in reproducing the `socket error`. `To try to run e3sm_diags in parallel on tc_analysis`, does it mean, when tc_analysis task is on, while `tc_analysis` set is not set to be one of `e3sm_diags` sets to run, so that `tc_analysis` and `e3sm-diags` run in parallel?

[the remark this answers is not in the document]
Comment (forsyth2) [reply zppy/e3sm_diags.py:36]: > I'm curious how `tc_analysis` dependency works differently than other dependencies such as climo and time-series.

In this pull request, it's this line in the task we're finishing up
```
carried_over_dependencies.append(statusFile)
```
and this line in the next task:
```
dependencies: List[str] = carried_over_dependencies
```
This resets all other dependencies, except for the one we specifically added to `carried_over_dependencies` (i.e., the last task).

>  I'm interested in reproducing the `socket error`

This error occurs intermittently when running two `e3sm_diags` jobs, both of which run the `tc_analysis` set. The sequential workaround was introduced in #169. A full discussion of the `tc_analysis` concurrency/parallelism bugs can be found in #180.

File: zppy/e3sm_diags.py [hunk 2]
```diff
@@ -99,7 +103,7 @@ def e3sm_diags(config: ConfigObj, script_dir: str, existing_bundles, job_ids_fil
                         # Note that this line should still be executed even if jobid == -1
                         # The later tc_analysis-using e3sm_diags tasks still depend on this task (and thus will also fail).
                         # Add to the dependency list
-                        dependencies.append(status_file)
+                        carried_over_dependencies.append(status_file)
```
Comment (forsyth2) [inline zppy/e3sm_diags.py:106]: Maybe rather than `carried_over_dependencies` we have a variable `previous_status_file` and then `if "tc_analysis" in c["sets"]` we add that to the dependencies as well. Then, diags not using tc_analysis wouldn't be affected.

**Comment by forsyth2** [comment]
Rebased after the modularizing refactor of #628. Merge conflicts affected:
```
	both modified:   zppy/e3sm_diags.py
	both modified:   zppy/global_time_series.py
	both modified:   zppy/ilamb.py
	both modified:   zppy/mpas_analysis.py
	both modified:   zppy/tc_analysis.py
```
and have now been resolved. It looks like the now 22 unit tests are passing.

I had to commit with `--no-verify` because of:
```
pre-commit run --all-files[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
An error has occurred: InvalidManifestError: 
==> File /gpfs/fs1/home/ac.forsyth2/.cache/pre-commit/repo0641jows/.pre-commit-hooks.yaml
==> At Hook(id='check-added-large-files')
==> At key: stages
==> At index 0
=====> Expected one of commit, commit-msg, manual, merge-commit, post-checkout, post-commit, post-merge, post-rewrite, prepare-commit-msg, push but got: 'pre-commit'
Check the log at /gpfs/fs1/home/ac.forsyth2/.cache/pre-commit/pre-commit.log
```
which I assume is related to #627.
```

**Comment by xylar** [comment]
@forsyth2, as I said in #627, I don't think this is an issue caused by that update alone.  I think a fresh install of `pre-commit` should take care of it since it worked fine for me on my laptop.

**Comment by forsyth2** [comment]
Yep, fresh-install worked. Thanks @xylar

File: zppy/ilamb.py [hunk 3]
```diff
@@ -31,8 +31,10 @@ def ilamb(config: ConfigObj, script_dir: str, existing_bundles, job_ids_file):
         return existing_bundles
 
     # --- Generate and submit ilamb scripts ---
-    dependencies: List[str] = []
     for c in tasks:
+
+        dependencies: List[str] = []
```
Comment (forsyth2) [inline zppy/ilamb.py:36]: All tasks that weren't changed in this PR already have `dependencies` reset inside the `for c in tasks`

File: zppy/mpas_analysis.py [hunk 4]
```diff
@@ -31,8 +31,13 @@ def mpas_analysis(config: ConfigObj, script_dir: str, existing_bundles, job_ids_
...
     # MPAS-Analysis task dependant on all previous ones. This may not
     # be 100% fool-proof, but should be a reasonable start
-    dependencies = []
+
+    # Dependencies carried over from previous task.
+    carried_over_dependencies: List[str] = []
+
     for c in tasks:
+
+        dependencies: List[str] = carried_over_dependencies
```
Comment (forsyth2) [inline zppy/mpas_analysis.py:40]: `mpas_analysis` doesn't have any other dependencies besides previous runs of itself, so we could just move this definition out of the `for c in tasks` loop, but I think it's more comprehensible to explicitly note the dependency on previous runs is being carried over, via the name `carried_over_dependencies`

**Comment by forsyth2** [comment]
The unit tests pass and I ran `zppy -c tests/integration/generated/test_min_case_carryover_dependencies_chrysalis.cfg`, checking the output directory for failed `status` files and checking the generated plots. Since `tc_analysis` currently doesn't work with v3 data and the `tc_analysis` dependency handling was tested in #631, I just took that out of this min-case test.

**Comment by forsyth2** [comment]
For reference, the dependency output from that min-case test:
```
$ cd /lcrc/group/e3sm/ac.forsyth2/zppy_min_case_carryover_dependencies_output/test-623-20241015-v4/v3.LR.historical_0051/post/scripts
$ grep "'dependencies'" *settings
```
We can then find the following:
```
e3sm_diags_atm_monthly_180x360_aave_model_vs_obs_1987-1988.settings:  'dependencies': [ '.../climo_atm_monthly_180x360_aave_1987-1988.status'],

e3sm_diags_atm_monthly_180x360_aave_mvm_model_vs_model_1987-1988_vs_1985-1986.settings:  'dependencies': [ '.../climo_atm_monthly_180x360_aave_1987-1988.status'],

global_time_series_1985-1995.settings: 'dependencies': [ '.../ts_atm_monthly_glb_1985-1989-0005.status',
                    '.../ts_atm_monthly_glb_1990-1994-0005.status',
                    '.../ts_lnd_monthly_glb_1985-1989-0005.status',
                    '.../ts_lnd_monthly_glb_1990-1994-0005.status',
                    '.../mpas_analysis_ts_1985-1989_climo_1985-1989.status',
                    '.../mpas_analysis_ts_1985-1995_climo_1990-1995.status'],

ilamb_1985-1988.settings: 'dependencies': [ '.../ts_land_monthly_1985-1986-0002.status',
                    '.../ts_land_monthly_1987-1988-0002.status',
                    '.../ts_atm_monthly_180x360_aave_1985-1986-0002.status',
                    '.../ts_atm_monthly_180x360_aave_1987-1988-0002.status'],

mpas_analysis_ts_1985-1989_climo_1985-1989.settings:  'dependencies': [],

mpas_analysis_ts_1985-1995_climo_1990-1995.settings:  'dependencies': [ '.../mpas_analysis_ts_1985-1989_climo_1985-1989.status'],
```
